tt_metal/api/device.hpp

```diff
@@ -0,0 +1,342 @@
+ */
+std::vector<CoreCoord> GetEthernetSockets(Device device, chip_id_t connected_chip_id);
+
+/**
+ * @brief Returns the number of banks for a specific buffer type on the device.
+ *
+ * @param device The device to query.
+ * @param buffer_type The type of buffer.
+ * @return Number of banks.
+ */
+uint32_t GetNumBanks(Device device, const BufferType &buffer_type);
+
+/**
+ * @brief Computes the offset of a specific bank for a buffer type on the device.
+ *
+ * @param device The device to query.
+ * @param buffer_type The type of buffer.
+ * @param bank_id The ID of the bank.
+ * @return Offset of the bank.
+ */
+int32_t GetBankOffset(Device device, BufferType buffer_type, uint32_t bank_id);
+
+/**
+ * @brief Retrieves bank IDs associated with a logical core for a given buffer type.
+ *
+ * @param device The device to query.
+ * @param buffer_type The type of buffer.
+ * @param logical_core The logical core coordinate.
+ * @return Reference to a vector of bank IDs.
+ */
+const std::vector<uint32_t> &BankIdsFromLogicalCore(Device device, BufferType buffer_type, const CoreCoord &logical_core);
+
+/**
+ * @brief Deallocates all buffers on the device.
+ */
+void DeallocateBuffers(Device device);
+
+
+/**
+ * @brief Retrieves the machine epsilon for the SFPU on the device.
+ *
+ * @param device The device to query.
+ * @return SFPU machine epsilon.
+ */
+float GetSfpuEps(Device device);
+
+/**
+ * @brief Retrieves the representation of NaN for the SFPU on the device.
+ *
+ * @param device The device to query.
+ * @return SFPU NaN value.
+ */
+float GetSfpuNan(Device device);
+
+/**
+ * @brief Retrieves the representation of infinity for the SFPU on the device.
+ *
+ * @param device The device to query.
+ * @return SFPU infinity value.
+ */
+float GetSfpuInf(Device device);
+
+/**
+ * @brief Retrieves a command queue for the given device and queue ID.
+ *
+ * @param device The device to query.
+ * @param cq_id The command queue ID (default: 0).
+ * @return CommandQueue handle.
+ */
+CommandQueue GetCommandQueue(Device device, size_t cq_id = 0);
+
+/**
+ * @brief Begins tracing for a specific command queue ID and trace ID on the device.
+ *
+ * @param device The device to query.
+ * @param cq_id The command queue ID.
+ * @param tid The trace ID.
+ */
+void BeginTrace(Device device, uint8_t cq_id, uint32_t tid);
+
+/**
+ * @brief Ends tracing for a specific command queue ID and trace ID on the device.
+ *
+ * @param device The device to query.
+ * @param cq_id The command queue ID.
+ * @param tid The trace ID.
+ */
+void EndTrace(Device device, uint8_t cq_id, uint32_t tid);
+
+/**
+ * @brief Replays a trace for a specific command queue ID and trace ID on the device.
+ *
+ * @param device The device to query.
+ * @param cq_id The command queue ID.
+ * @param tid The trace ID.
+ * @param blocking Indicates whether the operation is blocking.
+ */
+void ReplayTrace(Device device, uint8_t cq_id, uint32_t tid, bool blocking);
```

Comment:
Should blocking/non-blocking be separate functions rather than a runtime parameter?

Reply:
Per discussion, we defer this.